packages/react-core/src/components/DatePicker/DatePicker.tsx

```diff
@@ -98,7 +98,7 @@ const DatePickerBase = (
     className,
     locale = undefined,
     dateFormat = yyyyMMddFormat,
-    dateParse = (val: string) => val.split('-').length === 3 && new Date(`${val}T00:00:00`),
+    dateParse = (val: string) => (val.split('-').length === 3 ? new Date(`${val}T00:00:00`) : new Date(undefined)),
```

Comment:
This resolves the issue mentioned here https://github.com/patternfly/patternfly-react/issues/9721#issuecomment-1752874585 regarding the return type of this prop. This basically just ensures that a Date type is always returned without really changing the outcome (instead of false we just get an invalid date object which should have the same effect).

Reply:
this should work fine - i'm approving as is. but i do think the point about forward slashes is valid, and if we don't add it as part of this, it probably makes sense to capture it in another issue.